pkg/netpol/eval/resources.go

```diff
@@ -422,6 +436,21 @@ func (pe *PolicyEngine) insertNetworkPolicy(np *netv1.NetworkPolicy) error {
 	return err
 }
 
+func (pe *PolicyEngine) insertAdminNetworkPolicy(anp *apisv1a.AdminNetworkPolicy) error {
+	// @TBD : currently disabling exposure-analysis when there are admin-network-policies in the input resources
+	if pe.exposureAnalysisFlag {
+		return errors.New(netpolerrors.ExposureAnalysisDisabledWithANPs)
+	}
+	if anp.Name == "" {
```

Comment:
no, I have no idea why; names are used to store the objects, and an object without a name would cause wrong results.
maybe, since our tests was always from `dir` the `yaml` correctness was assumed to be.
I had the need to check the names after the missing names of objects in new tests added by Tanya caused unexpected results.

Reply:
can try adding a badpath test with policy without name?
where in the flow does it cause an issue if admin policy has no name? if it is common also for regular netpol, can try debug that run

Reply:
actually,
tests that failed were tests with more than one ANP when all policies without names, so the policy-engine acted as if only one policy was provided only; the problem was not the empty name string but the uniqueness of the names.

1. when we have a single policy, we will not have wrong results (it will be stored in the map as `{"":&anp}` (with empty string as its name).
but when we have 2 policies with empty names, only one will be stored in the map with this empty string key and then we'll have unexpected results
2. according to [this](https://kubernetes.io/docs/concepts/overview/working-with-objects/names#names) names must be unique 

so maybe the next check `if there is already a policy with this name` is enough?

Reply:
yes, so better check for unique names (namespace + name) for both regular netpols and admin netpols

Reply:
done